src/go/rpk/pkg/vcloud/login_test.go

```diff
@@ -0,0 +1,74 @@
+// Copyright 2021 Vectorized, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+
+package vcloud_test
+
+import (
+	"testing"
+
+	"github.com/vectorizedio/redpanda/src/go/rpk/pkg/vcloud"
+)
+
+func TestLogin(t *testing.T) {
+	var tests = []struct {
+		name           string
+		tokenResponse  *vcloud.TokenResponse
+		tokenError     error
+		deviceResponse *vcloud.DeviceCodeResponse
+		deviceError    error
+
+		expectedError bool
```

Comment:
I've found it's valuable to expect a specific error message, since sometimes the function might fail, but not for the reasons you'd expect.